### ISSUE TITLE
ApplicationFee amount is not accepted, eventhough it is setup as Amount object

### ISSUE THREAD

**Issue by RickvF** [Viincenttt/MollieApi#396]
I have setup an OrderRequest with ApplicationFee as follows:

ApplicationFee = new ApplicationFee()
{
    Amount = new Amount(Currency.EUR, "0.05"),
    Description = "Test Fee"
},

When I execute the OrderResponse result = await orderClient.CreateOrderAsync(orderRequest); I get the following result in the browser.

I have even tried the same value as I use in the OrderRequest.Amount, however, this is still showing me the same exception with a different amount.
and I tried
ApplicationFee = new ApplicationFee()
{
    Amount = new Amount(Currency.EUR, 0.05m),
    Description = "Test Fee"
},

Can it be that there is an issue with how ApplicationFee is processed?

Project details:
.NET 7.0
Mollie.Api version 4.4.0

An unhandled exception occurred while processing the request.
MollieApiException: Unprocessable Entity - Parameters with a dot should be passed as objects. Did you mean: '{"payment": {"applicationFee": {"amount":{"currency":"EUR","value":"0.05"},"description":"Test Fee"}}}'?

Mollie.Api.Client.BaseMollieClient.ProcessHttpResponseMessage<T>(HttpResponseMessage response)

    Stack Query Cookies Headers Routing 

    MollieApiException: Unprocessable Entity - Parameters with a dot should be passed as objects. Did you mean: '{"payment": {"applicationFee": {"amount":{"currency":"EUR","value":"0.05"},"description":"Test Fee"}}}'?
        Mollie.Api.Client.BaseMollieClient.ProcessHttpResponseMessage<T>(HttpResponseMessage response)
        Mollie.Api.Client.BaseMollieClient.SendHttpRequest<T>(HttpMethod httpMethod, string relativeUri, object data)
        Mollie.Api.Client.BaseMollieClient.PostAsync<T>(string relativeUri, object data)
        Mollie.Api.Client.OrderClient.CreateOrderAsync(OrderRequest orderRequest)
        OrderResponse result = await orderClient.CreateOrderAsync(orderRequest);

**Comment by RickvF** [comment]
I have investigated this myself further. Hereby are my findings:

In Mollie.Api.Models.Order.Request the `ApplicationFee` attribute should not be part of the `OrderRequest` class as `[JsonProperty("payment.applicationFee")]` will not be translated to `Payment` Object and within an `ApplicationFee` object but to a `Payment.ApplicationFee` object.
Therefore I think and propose that the `ApplicationFee` becomes an attribute of Mollie.Api.Models.Order.Request.PaymentSpecificParameters since the `PaymentSpecificParameters` is already JSON converted to the `Payment` object for the order.

When cloning the code, modifying the above changes, and executing the request gives me the following error

               MollieApiException: Unprocessable Entity - Unable to process application request for this account

               Mollie.Api.Client.BaseMollieClient.ProcessHttpResponseMessage<T>(HttpResponseMessage response) in BaseMollieClient.cs

                throw new MollieApiException(errorDetails);

                Mollie.Api.Client.BaseMollieClient.SendHttpRequest<T>(HttpMethod httpMethod, string relativeUri, object data)
                Mollie.Api.Client.BaseMollieClient.PostAsync<T>(string relativeUri, object data) in BaseMollieClient.cs

                return await SendHttpRequest<T>(HttpMethod.Post, relativeUri, data).ConfigureAwait(false);

                Mollie.Api.Client.OrderClient.CreateOrderAsync(OrderRequest orderRequest) in OrderClient.cs

                return await PostAsync<OrderResponse>("orders", orderRequest).ConfigureAwait(false);


I started to wonder if this issue is related that I am using my test APIKEY as well as that I am authorizing the same account that has created the OAuth application as the account to use to create the order.
Unfortunately, I do not have a second organization yet live to test this out, but maybe you can comment on this.


In case the issue I am now facing is related to using the same organization as the app creator and authenticated organization or that I am using a test APIKEY. When that is solved and if you like the above-suggested solution of having the `ApplicationFee` as part of the `PaymentSpecificParameters` I can submit a Pull Request

**Comment by Viincenttt** [comment]
Hi @RickvF ,

Thanks for reporting this. This does seem to be a bug in the .NET library, where the applicationFee is passed in the incorrect JSON format. Currently, it is sending:
```
{
   ...
    "locale": "nl_NL",
    "orderNumber": "16738",
    "payment.applicationFee": {
        "amount": {
            "currency": "EUR",
            "value": "0.25"
        },
        "description": "test"
    },
    ....
}
```

According to the error you mentioned, it should be:
```
{
   ...
    "locale": "nl_NL",
    "orderNumber": "16738",
    "payment": {
       "applicationFee": {
          "amount": {
              "currency": "EUR",
              "value": "0.25"
          },
          "description": "test"
	   }
    },
    ....
}
```

I can fix this by moving it to the `PaymentSpecificParameters` object like you mentioned. 

Regarding the `Unable to process application request for this account` error, I suggest contacting Mollie as it sounds like an account specific issue.

**Comment by RickvF** [comment]
@Viincenttt  That would be awesome if you can do this update!

Let me contact Mollie, thanks

**Comment by Viincenttt** [comment]
This is now fixed in version 4.4.1, which is available on NuGet :)

**Comment by RickvF** [comment]
Thanks!!
